js/layer_manager_row.ts

```diff
@@ -0,0 +1,238 @@
+                0% { transform: rotate(0deg); }
+                100% { transform: rotate(360deg); }
+            }
+
+            button.loading .spinner,
+            button.loading:hover .close-icon,
+            button.done-loading .close-icon {
+                display: block;
+            }
+
+            button.loading .close-icon,
+            button.loading:hover .spinner,
+            button.done-loading .spinner {
+                display: none;
+            }
+
+            .remove-layer-text {
+                flex-grow: 1;
+            }
+
+            .confirm-deny-button {
+                height: 26px;
+                width: 70px;
+            }
+        `,
+    ];
+
+    private _model: AnyModel<LayerManagerRowModel> | undefined = undefined;
+    private static modelNameToViewName = new Map<keyof LayerManagerRowModel, keyof LayerManagerRow | null>([
+        ["name", "name"],
+        ["visible", "visible"],
+        ["opacity", "opacity"],
+        ["is_loading", "isLoading"],
+    ]);
+
+    set model(model: AnyModel<LayerManagerRowModel>) {
+        this._model = model;
+        for (const [modelKey, widgetKey] of LayerManagerRow.modelNameToViewName) {
+            if (widgetKey) {
+                // Get initial values from the Python model.
+                (this as any)[widgetKey] = model.get(modelKey);
+                // Listen for updates to the model.
+                model.on(`change:${modelKey}`, () => {
+                    (this as any)[widgetKey] = model.get(modelKey);
+                });
+            }
+        }
+    }
+
+    @property()
+    name: string = "";
+
+    @property()
+    visible: boolean = true;
+
+    @property()
+    opacity: number = 0;
+
+    @property()
+    isLoading: boolean = false;
+
+    @property()
+    isConfirmDialogVisible: boolean = false;
+
+    render(): TemplateResult {
+        return html`
+            <div class="row">
+                <input
+                    type="checkbox"
+                    class="layer-visibility-checkbox"
+                    .checked="${this.visible}"
+                    @change="${this.onLayerVisibilityChanged}"
+                />
+                <span class="legacy-text layer-name">${this.name}</span>
+                <input
+                    type="range"
+                    class="legacy-slider layer-opacity-slider"
+                    min="0"
+                    max="1"
+                    step="0.01"
+                    .value="${this.opacity}"
+                    @input="${this.onLayerOpacityChanged}"
+                />
+                <button
+                    class="legacy-button settings-delete-button"
+                    @click="${this.onSettingsClicked}"
+                >
+                    <span class="material-symbols-outlined">&#xe8b8;</span>
+                </button>
+                <button
+                    class="legacy-button settings-delete-button ${this.isLoading ? 'loading' : 'done-loading'}"
```

Comment:
use `classMap` directive for a better time: https://lit.dev/docs/templates/directives/#classmap

Reply:
Done.